modules/web-discovery-project/sources/remote-resource-watcher.ts

```diff
@@ -183,11 +186,14 @@ export class RemoteResourceWatcher {
     this.moduleName = nonEmpty(options.moduleName, "moduleName");
     this.resourceUrl = nonEmpty(options.resource.url, "resource.url");
 
-    this.signatureUrl = nonEmpty(options.signature.url, "signature.url");
-    this.verifier = nonEmpty(options.signature.verifier, "signature.verifier");
+    if (!options.insecure) {
```

Comment:
Nit. I would initialize `this.insecure = options.insecure || false;` earlier and use it in this condition.